util/bloom_test.cc

```diff
@@ -458,6 +473,172 @@ TEST_F(FullBloomTest, Schema) {
   ResetPolicy();
 }
 
+struct RawFilterTester {
+  std::array<char, 3000> data_;
+  Slice cur_slice_;
+  char* metadata_ptr_;
+
+  RawFilterTester() : metadata_ptr_(&data_.back() - 4) {}
+
+  Slice& ResetNoFill(uint32_t len_without_metadata, uint32_t num_lines,
```

Comment:
Not a big deal but I feel that avoiding a copy of Slice in unit tests not worth it and complicates the logic unnecessarily. I suggest we plainly return Slice here.